File: 8주차/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    answer = True
    phone_book=sorted(phone_book, key=lambda x: (x,len))

    for i in range(0, len(phone_book)-1):
        if phone_book[i]==phone_book[i+1][:len(phone_book[i])]:
            answer=False
            break
    return answer

File: 8주차/42577. 전화번호 목록/전화번호 목록.java
import java.util.Arrays;
import java.util.Comparator;
import java.util.HashMap;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        HashMap<String, Integer> hashMap = new HashMap<>();
        for(int i = 0; i < phone_book.length; i++) {
			hashMap.put(phone_book[i], 1);
		}

		
		first: for(int i = 0; i < phone_book.length; i++) {
			for(int j = 1; j < phone_book[i].length(); j++) {
				if(hashMap.containsKey(phone_book[i].substring(0, j))) {
					answer = false;
					break first;
				}
			}
		}
        return answer;
    }
}